group-coordinator/src/main/java/org/apache/kafka/coordinator/group/modern/ModernGroup.java

```diff
@@ -367,18 +366,15 @@ public void setMetadataHash(long metadataHash) {
     public static long computeMetadataHash(
         Map<String, SubscriptionCount> subscribedTopicNames,
         Map<String, Long> topicHashCache,
-        MetadataImage metadataImage
+        CoordinatorMetadataImage metadataImage
     ) {
         Map<String, Long> topicHash = new HashMap<>(subscribedTopicNames.size());
-        subscribedTopicNames.keySet().forEach(topicName -> {
-            TopicImage topicImage = metadataImage.topics().getTopic(topicName);
-            if (topicImage != null) {
+        subscribedTopicNames.keySet().forEach(topicName ->
```

Comment:
Now that we've added the ifPresent in there I'd suggest writing it fully in streams, e.g.

```
        Map<String, Long> topicHash = subscribedTopicNames.keySet().stream()
                .filter(topicName -> metadataImage.topicMetadata(topicName).isPresent())
                .collect(Collectors.toMap(
                        topicName -> topicName,
                        topicName -> topicHashCache.computeIfAbsent(topicName, k -> Utils.computeTopicHash(k, metadataImage))));
```

Reply:
nice yea that's better